torch/csrc/jit/python/init.cpp

```diff
@@ -1388,15 +1388,37 @@ void initJITBindings(PyObject* module) {
           if (size == 0) {
             return size;
           }
-          py::gil_scoped_acquire acquire;
-          auto memory_view = py::memoryview::from_memory(
-              reinterpret_cast<const char*>(data), size);
-          buffer.attr("write")(std::move(memory_view));
+          if (!data) {
+            // See [Note: write_record_metadata]
+            buffer.attr("seek")(
+                size, py::module::import("os").attr("SEEK_CUR"));
+          } else {
+            py::gil_scoped_acquire acquire;
```

Comment:
You need the gil for any python api! So the code above needs it as well!

Reply:
Oops, very much my bad! Thanks for catching this, fixed